swig/StringArray_API_extensions.i

```diff
@@ -0,0 +1,281 @@
+    {
+        std::for_each(_array.begin(), _array.end(), [](char* c) { delete[] c; });
+    }
+
+    const size_t _string_size;
+    std::vector<char*> _array;
+};
+
+
+
+
+
+
+
+    /**
+     * Return the pointer to the array of strings.
+     * Wrapped in Java into String[] automatically.
+     */
+    char **StringArrayHandle_get_strings(StringArrayHandle handle)
+    {
+        return reinterpret_cast<StringArray *>(handle)->data();
+    }
+
+    /**
+     * For the end user to extract a specific string from the StringArray object.
+     */
+    char *StringArrayHandle_get_string(StringArrayHandle handle, int feature_idx)
+    {
+        return reinterpret_cast<StringArray *>(handle)->getitem(feature_idx);
+    }
+
+    /**
+     * Free the StringArray object.
+     */
+    void StringArrayHandle_free(StringArrayHandle handle)
+    {
+        delete reinterpret_cast<StringArray *>(handle);
+    }
+
+    /**
+     * Retrieve the number of strings in the StringArray.
+     */
+    int StringArrayHandle_get_num_elements(StringArrayHandle handle)
+    {
+        return reinterpret_cast<StringArray *>(handle)->get_num_elements();
+    }
+
+    /**
+     * @brief Wraps LGBM_BoosterGetEvalNames.
+     *
+     * In case of success a new StringArray is created and returned,
+     * which you're responsible for freeing,
+     * @see StringArrayHandle_free().
+     * In case of failure such resource is freed and nullptr is returned.
+     * Check for that case with null (lightgbmlib) or 0 (lightgbmlibJNI).
+     *
+     * @param handle Booster handle
+     * @param eval_counts number of evaluations
+     *
+     * @return StringArrayHandle with the eval names (or nullptr in case of error)
+     */
+    StringArrayHandle LGBM_BoosterGetEvalNamesSWIG(BoosterHandle handle,
+                                                   int eval_counts)
+    {
+        std::unique_ptr<StringArray> strings(nullptr);
+
+        try {
+            // TODO: @reviewer, 128 was the chosen size, any particular reason for this constraint?
+            strings.reset(new StringArray(eval_counts, 128));
+        } catch (std::bad_alloc &e) {
+            LGBM_SetLastError("Failure to allocate memory."); // TODO: @reviewer - why is this not setting the message?
+            return nullptr;
+        }
+
+        API_OK_OR_NULL(LGBM_BoosterGetEvalNames(handle, &eval_counts, strings->data()));
+        return strings.release();
+    }
+
+    /**
+     * @brief Wraps LGBM_BoosterGetFeatureNames.
+     *
+     * Allocates a new StringArray. You must free it yourself if it succeeds.
+     * @see StringArrayHandle_free().
+     * In case of failure such resource is freed and nullptr is returned.
+     * Check for that case with null (lightgbmlib) or 0 (lightgbmlibJNI).
+     *
+     * @param handle Booster handle
+     * @return StringArrayHandle with the feature names (or nullptr in case of error)
+     */
+    StringArrayHandle LGBM_BoosterGetFeatureNamesSWIG2(BoosterHandle handle)
```

Comment:
let's just keep it as LGBM_BoosterGetFeatureNamesSWIG?

Reply:
You're perfectly right! Great catch! I'm sorry, was doing some experimentation and forgot to put that back.